components/post_view/post_info/post_info.tsx

```diff
@@ -21,116 +22,125 @@ import PostReaction from 'components/post_view/post_reaction';
 import PostTime from 'components/post_view/post_time';
 import InfoSmallIcon from 'components/widgets/icons/info_small_icon';
 
-export default class PostInfo extends React.PureComponent {
-    static propTypes = {
-
-        /*
-         * The post to render the info for
-         */
-        post: PropTypes.object.isRequired,
-
-        /*
-         * The id of the team which belongs the post
-         */
-        teamId: PropTypes.string,
-
-        /*
-         * Function called when the comment icon is clicked
-         */
-        handleCommentClick: PropTypes.func.isRequired,
-
-        /*
-         * Function called when the card icon is clicked
-         */
-        handleCardClick: PropTypes.func.isRequired,
-
-        /*
-         * Funciton called when the post options dropdown is opened
-         */
-        handleDropdownOpened: PropTypes.func.isRequired,
-
-        /*
-         * Set to mark the post as flagged
-         */
-        isFlagged: PropTypes.bool,
-
-        /*
-         * Set to mark the post as open the extra info in the rhs
-         */
-        isCardOpen: PropTypes.bool,
-
-        /*
-         * The number of replies in the same thread as this post
-         */
-        replyCount: PropTypes.number,
+export type Props = {
+
+    /**
+     * The post to render the info for
+     */
+    post: any;
```

Comment:
This should be of type `Post` from `mattermost-redux`

Reply:
@devinbinnie - when I tried to change from `any` to `Post`, there are lots of errors. One such error is:
```
components/post_view/post_info/post_info.test.tsx:53:34 - error TS2322: Type '{ post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }; ... 12 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is missing the following properties from type 'Post': reply_count, metadata

53         const wrapper = shallow(<PostInfo {...requiredProps}/>);
                                                          ~~~~~~~~
```

Reply:
Since the test file is still consuming the component, it expects the test to provide all of the fields in the `Post` object, which it might not be.

A couple ways to solve this:
1) I think you can cast the `post` prop as `any` (only for the test)
2) You can mock out the entire post object.

Reply:
Thanks for responding @devinbinnie . Here's the stacktraces of errors:
```
components/post_view/post_info/post_info.test.tsx:53:34 - error TS2322: Type '{ post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }; ... 12 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is missing the following properties from type 'Post': reply_count, metadata

53         const wrapper = shallow(<PostInfo {...requiredProps}/>);
                                    ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:60:34 - error TS2322: Type '{ compactDisplay: boolean; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; ... 6 more ...; user_id: string; }; ... 11 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

60         const wrapper = shallow(<PostInfo {...props}/>);
                                    ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:67:34 - error TS2322: Type '{ useMilitaryTime: boolean; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; ... 6 more ...; user_id: string; }; ... 11 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

67         const wrapper = shallow(<PostInfo {...props}/>);
                                    ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:74:34 - error TS2322: Type '{ isFlagged: boolean; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; ... 5 more ...; user_id: string; }; ... 11 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

74         const wrapper = shallow(<PostInfo {...props}/>);
                                    ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:82:34 - error TS2322: Type '{ post: { is_pinned: boolean; channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }; ... 12 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ is_pinned: boolean; channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is missing the following properties from type 'Post': reply_count, metadata

82         const wrapper = shallow(<PostInfo {...requiredPropsWithPinnedPost}/>);
                                    ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:90:34 - error TS2322: Type '{ post: { type: string; channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; ... 4 more ...; user_id: string; }; ... 12 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ type: string; channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; ... 4 more ...; user_id: string; }' is missing the following properties from type 'Post': reply_count, metadata

90         const wrapper = shallow(<PostInfo {...requiredPropsWithEphemeralPost}/>);
                                    ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:98:34 - error TS2322: Type '{ post: { state: string; channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; ... 5 more ...; user_id: string; }; ... 12 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ state: string; channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; ... 5 more ...; user_id: string; }' is missing the following properties from type 'Post': reply_count, metadata

98         const wrapper = shallow(<PostInfo {...requiredPropsWithDeletedPost}/>);
                                    ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:106:34 - error TS2322: Type '{ post: { type: string; state: string; channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; ... 5 more ...; user_id: string; }; ... 12 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ type: string; state: string; channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; ... 4 more ...; user_id: string; }' is missing the following properties from type 'Post': reply_count, metadata

106         const wrapper = shallow(<PostInfo {...requiredPropsWithDeletedEphemeralPost}/>);
                                     ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:112:14 - error TS2322: Type '{ enableEmojiPicker: true; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; ... 6 more ...; user_id: string; }; ... 11 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

112             <PostInfo
                 ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:124:34 - error TS2322: Type '{ handleDropdownOpened: Mock<any, any>; enableEmojiPicker: boolean; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; ... 8 more ...; user_id: string; }; ... 10 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

124         const wrapper = shallow(<PostInfo {...requiredPropsWithHandleDropdownOpened}/>);
                                     ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:137:34 - error TS2322: Type '{ actions: { removePost: Mock<any, any>; }; enableEmojiPicker: boolean; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; ... 8 more ...; user_id: string; }; ... 10 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

137         const wrapper = shallow(<PostInfo {...requiredPropsWithRemovePost}/>);
                                     ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:147:34 - error TS2322: Type '{ hover: boolean; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; ... 5 more ...; user_id: string; }; ... 11 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

147         const wrapper = shallow(<PostInfo {...props}/>);
                                     ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:154:34 - error TS2322: Type '{ showTimeWithoutHover: boolean; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; ... 6 more ...; user_id: string; }; ... 11 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

154         const wrapper = shallow(<PostInfo {...props}/>);
                                     ~~~~~~~~

components/post_view/post_info/post_info.test.tsx:165:14 - error TS2322: Type '{ isFlagged: boolean; post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; ... 5 more ...; user_id: string; }; ... 11 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is not assignable to type 'Post'.

165             <PostInfo {...props}/>,
                 ~~~~~~~~


Found 14 errors.
```

Reply:
Yep, looks like you need to do one of the above solutions.
If you need to mock out the whole `Post` object, consider adding the mock function to `utils/test_helper`.

Reply:
> Yep, looks like you need to do one of the above solutions.
> If you need to mock out the whole `Post` object, consider adding the mock function to `utils/test_helper`.

@devinbinnie I needed this `getPostMock` util function while working on https://github.com/mattermost/mattermost-webapp/pull/6717, so maybe if my PR got reviewed and merged before this one, @vraravam can merge master and use it here as well.

Reply:
Yes - that would be super helpful @abdusabri - thanks!

Reply:
@vraravam The PR https://github.com/mattermost/mattermost-webapp/pull/6717 is merged, you can merge master and use the mock utility function here 🙂

Reply:
@abdusabri - thanks. I have fixed all the issues (except 2 failing tests) - for which I need some help. The branch has been rebased from `master` to bring in your changes to the `TestHelper`. Please do help if you can.

Reply:
@vraravam I think if you cast that field in the test helper as `any` that should fix the issue. TS is weird with those specific string union types.

Reply:
nope @devinbinnie - that is still broken

Reply:
@vraravam casting `Posts.POST_DELETED` as literal `'DELETED'` solves the issue 👇 

![image](https://user-images.githubusercontent.com/25670682/96155168-ea9cc200-0f0f-11eb-99e5-0c13caab0052.png)

BTW, since the `state` is optional in the `Post` type, there is no need to add it to the mock utility function.

cc: @devinbinnie

Reply:
Thanks @abdusabri!

Reply:
still no luck @abdusabri 
<img width="905" alt="Screen Shot 2020-10-16 at 10 01 54 PM" src="https://user-images.githubusercontent.com/69629/96284455-3710fc00-0ffb-11eb-8853-b32dede7452c.png">

Reply:
finally solved it with another contributor (thanks @karuppiah7890)

@abdusabri - this is now ready to be reviewed and merged.

Reply:
@vraravam, I'm afraid that the same error about Post State Type still exists. 👇 

![image](https://user-images.githubusercontent.com/25670682/96334747-4fbcf880-1073-11eb-98fb-8182a61f8060.png)

Have you checked this comment? https://github.com/mattermost/mattermost-webapp/pull/6619#discussion_r505658647

Anyway, I'm just a fellow contributor 🙂 (not a Mattermost organization member), So I can't help with approving or merging PRs

Reply:
thanks for clarifying @abdusabri  - your help is very much appreciated.

On the matter of the `make check-types` - it is passing on my machine against the `HEAD` of the branch.
<img width="1093" alt="Screen Shot 2020-10-17 at 4 29 32 PM" src="https://user-images.githubusercontent.com/69629/96335416-13e75a80-1096-11eb-9c48-48ddc13d88de.png">

Reply:
@devinbinnie - not sure whether its an issue on the CI side or something else. I have done a fresh clone of the whole repo, deleted all `__snapshots__` folders, and rerun the `check-types` and `test` make targets - but, on my local (as shown in the above pic), there is no error raised. Need some help - since one of the changes in this PR is a pre-requisite for another PR that I created last night (related to `dot_menu`) - for which I would like to resolve the conflicts while both of them are fresh in my mind.

Reply:
@vraravam I'll merge in master which should kick the CI and we'll see if that fixes it. If not, pull down your branch (with the merged master branch) and we'll see if the error persists.

If it does, we can just cast it as an `any` and be done with it for now. Sorry this has been such a hassle.

Reply:
thanks @devinbinnie for helping break the impasse.

Reply:
@devinbinnie - was this merged into `master`? Can we merge the PR if not?